packages/block-editor/src/store/selectors.js

```diff
@@ -1236,11 +1236,22 @@ export function isBlockSelected( state, clientId ) {
  * @return {boolean} Whether the block has an inner block selected
  */
 export function hasSelectedInnerBlock( state, clientId, deep = false ) {
-	return getBlockOrder( state, clientId ).some(
-		( innerClientId ) =>
-			isBlockSelected( state, innerClientId ) ||
-			isBlockMultiSelected( state, innerClientId ) ||
-			( deep && hasSelectedInnerBlock( state, innerClientId, deep ) )
+	const selectedBlockClientIds = getSelectedBlockClientIds( state );
+
+	if ( ! selectedBlockClientIds.length ) {
+		return false;
+	}
+
+	if ( deep ) {
+		return selectedBlockClientIds.some( ( id ) =>
+			// Pass true because we don't care about order and it's more
+			// performant.
+			getBlockParents( state, id, true ).includes( clientId )
```

Comment:
I wonder if we can stop this as a single block. (Maybe we can if the current block clientId is not selected)

Reply:
Not sure what you mean, do you mean merging with the non deep code block?

Reply:
I mean if one single selected block is not within the `clientId`, probably the others won't be and we could just check `selectedBlockClientIds[ 0 ]` 

That said:

 - This only works if the `clientId` is not within the selection itself.
 - It is based on an assumption that non-consecutive selection is not supported.

Reply:
Let's keep it future proof then :)

Reply:
Yes, but not sure if it's valuable after the lack of a real improvement in the metric. The most important is that we don't run anything if no block is selected, and if there is, it's a pretty small loop. For multi selection yes, I guess we could just check the first ID. Not sure if we'd want to keep it future proof for non consecutive selection.